### ISSUE TITLE
Default Resolution of Video Device Not Alterable

### ISSUE THREAD

**Issue by 151GuidingLight** [qTox/qTox#3535]
##### Brief Description

There is no save feature for users to save default setting for video and audio input devices; frame rates are choppy as a result, though unknown if this is linked or separate problem.
OS: OS X 10.11.6
qTox version: OSX-Travis-R5
Commit hash: 886c96547467b91a0030b31d615a6dcefe45b2a5
toxcore: 0.0.0
Qt: 5.6.0
Hardware: Mac Mini (Mid 2011) and HD Pro Webcam C920 (by Logitech with pre-rendering to reduce system workload)

Reproducible: Always
##### Steps to reproduce
1. Click on Audio/Video tab
2. Watch as video device resolution is at 1080p
3. Watch as video frame rate is awful until user reduces it to 480p
##### Observed Behavior

User is unable to change default resolution of video input device, and it is stuck at 1080p, at least for me, which is very choppy.  I always have to switch down to 480p and there is no way to save this setting within qTox.  This was not during a video call (so encryption cannot be blamed, this was a local test).
##### Expected Behavior

It is expected that frame rates would not be choppy, my system CPU usage is low, my GPU isn't a problem or bottleneck, especially when Skype and Google Hangouts, which really heat up the system, have smooth frame rates with this same setup.

**Comment by zetok** [comment]
> There is no save feature for users to save default setting for video and audio input devices; (…)

Note – this most likely was fixed in commits since 1.4. osx build depends on #3520, which needs to be fixed before osx could get an update to 1.5.

> (…) frame rates are choppy as a result, though unknown if this is linked or separate problem.

And this is something that could be improved in both qTox and toxcore. Also, a ~separate issue.

There is irungentoo/toxcore#1606, dunno if it'll be merged in timely manner.

**Comment by BiTOk** [comment]
I have the same problem with video resolution in Windows 10 with 1.4.1 build.

**Comment by Diadlo** [comment]
Duplicate of #1033 ?
If so, fixed by #3371

**Comment by BiTOk** [comment]
It looks similar. Is there is test build with this fix?

**Comment by Diadlo** [comment]
@BiTOk As I know, you can test just manual build
@zetok Right?

**Comment by zetok** [comment]
Yes.

There also might be nightly builds for windows on jekins.

**Comment by BiTOk** [comment]
Last nightly build (ff92a5595002a4156bada85eeffe4bdb3cdd7f8e) works fine, issue does not reproduce.

ADD: But in resolutions list I see two 240p.

**Comment by zetok** [comment]
Since it's a duplicate of #1033, closing.

**Comment by seankhl** [comment]
Glad someone is fixing my awful first-pass best mode heurstic!
